Manager/BaseSalesManager.php

```diff
@@ -0,0 +1,143 @@
+
+        if (($isNew && !($keyExists && $data[$key] !== null)) || (!$keyExists || $data[$key] === null)) {
+            throw new \Exception(sprintf("Missing attribute %s", $key));
+        }
+
+        return $keyExists;
+    }
+
+    /**
+     * checks if data is set
+     *
+     * @param $key
+     * @param $data
+     *
+     * @return bool
+     */
+    protected function checkIfSet($key, $data)
+    {
+        $keyExists = array_key_exists($key, $data);
+
+        return $keyExists && $data[$key] !== null && $data[$key] !== '';
+    }
+
+    /**
+     * searches for contact in specified data and calls callback function
+     *
+     * @param array $data
+     * @param $dataKey
+     * @param $addCallback
+     *
+     * @return null|Contact
+     * @throws \Exception
+     */
+    protected function addContactRelation(array $data, $dataKey, $addCallback)
+    {
+        $contact = null;
+        if (array_key_exists($dataKey, $data) && is_array($data[$dataKey]) && array_key_exists('id', $data[$dataKey])) {
+            /** @var Contact $contact */
+            $contactId = $data[$dataKey]['id'];
+            $contact = $this->em->getRepository(static::$contactEntityName)->find($contactId);
+            if (!$contact) {
+                // TODO: custom exception dependencyNotFound
+                throw new \Exception(sprintf("Entity '%s' with id %s not found", static::$contactEntityName, $contactId));
+            }
+            $addCallback($contact);
+        }
+
+        return $contact;
+    }
+
+    /**
+     * Returns the entry from the data with the given key, or the given default value, if the key does not exist
+     *
+     * @param array $data
+     * @param string $key
+     * @param string $default
+     *
+     * @return mixed
+     */
+    protected function getProperty(array $data, $key, $default = null)
+    {
+        return array_key_exists($key, $data) ? $data[$key] : $default;
+    }
+
+    /**
+     * @param $data
+     * @param Order $order
+     *
+     * @return null|object
+     * @throws OrderDependencyNotFoundException
+     * @throws \Exception
+     */
+    protected function setAccount($data, $entity)
+    {
+        $accountData = $this->getProperty($data, 'account');
+        if ($accountData) {
+            if (!array_key_exists('id', $accountData)) {
+                // TODO: custom exception
+                throw new \Exception(sprintf("Missing attribute %s", 'account.id'));
+            }
+            // TODO: inject repository class
+            $account = $this->em->getRepository(static::$accountEntityName)->find($accountData['id']);
+            if (!$account) {
+                // TODO custom exception: dependencyNotFound
+                throw new \Exception(sprintf("Entity '%s' with id %s not found", static::$accountEntityName, $accountData['id']));
+            }
+            $entity->setAccount($account);
+
+            return $account;
+        } else {
```

Comment:
else is not needed ?
as code will return anyway in the if cause above you could simple add:

$entity->setAccount(null);
return null;